metron-platform/metron-common/src/main/java/org/apache/metron/common/configuration/manager/ZkConfigurationManager.java

```diff
@@ -0,0 +1,129 @@
+/*
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.metron.common.configuration.manager;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.NodeCache;
+import org.apache.curator.utils.CloseableUtils;
+import org.apache.metron.common.utils.JSONUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.commons.lang.ArrayUtils.isNotEmpty;
+
+/**
+ * Responsible for managing configuration values that are created, persisted, and updated
+ * in Zookeeper.
+ */
+public class ZkConfigurationManager implements ConfigurationManager {
```

Comment:
Thanks for the background @merrimanr ! 

> deserialization happens in a synchronized method every time a tuple is executed.

Good point about serialization.  Let me see if I can optimize that.  

> It's a bummer you couldn't reuse ConfiguredBolt. Is there a way we can make it more flexible so that you can reuse it?

`ConfiguredBolt` inherits from `BaseRichBolt` whereas I am using Storm's windowing functionality and need to inherit from `BaseWindowedBolt`.  

Having said that, I kind of prefer to delegate the functionality to a separate class like `ZkConfigurationMgr` as it can be easier to test and reuse.  `ConfiguredBolt` could delegate the logic to a `ZkConfigurationMgr`.  But that's just my $.02 after digging into this a bit.  I can see advantages in both ways of doing it.